controllers/managedocs_controller.go

```diff
@@ -1408,6 +1420,37 @@ func (r *ManagedOCSReconciler) removeOLMComponents() error {
 	return nil
 }
 
+func (r *ManagedOCSReconciler) removeStorageSystem() error {
+	storageSystemList := odfv1a1.StorageSystemList{}
+	if err := r.list(&storageSystemList); err != nil {
+		return fmt.Errorf("unable to list storageSystem resource: %v", err)
+	}
+	for i := range storageSystemList.Items {
+		storageSystem := storageSystemList.Items[i]
+		if err := r.delete(&storageSystem); err != nil {
+			return fmt.Errorf("unable to delete storageSystem: %v", err)
+		}
+	}
+	return nil
+}
+
+func (r *ManagedOCSReconciler) removeOCSCSV() error {
+	csvList := opv1a1.ClusterServiceVersionList{}
+	if err := r.list(&csvList); err != nil {
+		return fmt.Errorf("unable to list csv resources: %v", err)
+	}
+
+	csv := getCSVByPrefix(csvList, ocsOperatorName)
+	if csv != nil {
+		if err := r.delete(csv); err != nil {
+			return fmt.Errorf("Unable to delete csv: %v", err)
+		}
+	}
+
+	r.Log.Info("OCS csv removed successfully")
+	return nil
+}
```

Comment:
This is going to be the second call sire where we invoke `getCSVByPrefix`. In both cases, the code first lists all CSV. I think we should fold the CSV list operation into the util (same name) then inline this function `removeOCSCSV` in the call site (as it is going to be 5 line function that is invoked only once